docs/administration/configuration/configuring-external-access/index.md

```diff
@@ -0,0 +1,136 @@
+You can find an example of custom Ingress below.
+
+??? note "Example of custom Ingress"
+
+    ```{ .yaml .copy-to-clipboard }
+    apiVersion: networking.k8s.io/v1
+    kind: Ingress
+    metadata:
+      name: reana-ingress
+      annotations:
+        kubernetes.io/ingress.class: traefik
+        traefik.ingress.kubernetes.io/router.entrypoints: web,websecure
+    spec:
+      tls:
+        - secretName: <tls-secret>
+      rules:
+        - http:
+            paths:
+            - path: /api
+              pathType: Prefix
+              backend:
+                service:
+                  name: <helm-release-prefix>-server
+                  port:
+                    number: 80
+            - path: /oauth
+              pathType: Prefix
+              backend:
+                service:
+                  name: <helm-release-prefix>-server
+                  port:
+                    number: 80
+            - path: /
+              pathType: Prefix
+              backend:
+                service:
+                  name: <helm-release-prefix>-ui
+                  port:
+                    number: 80
+          host: <hostname>
+    ```
+
+## Reverse proxy
+
+You might be interested in placing a (reverse) proxy in front of REANA, but some additional configuration might be needed to make REANA work correctly.
+
+Let's consider a setup in which a generic proxy handles connections to REANA, like so:
+
+```text
+          +---------+    +---------+
+Client -> |  Proxy  | -> |  REANA  |
+          +---------+    +---------+
+```
+
+In this case, REANA has no way to know how the client connected to the server, as REANA will only see connections coming from the proxy.
+If the proxy also performs TLS termination, all the connections made to REANA will be over HTTP, even if the client connected over HTTPS.
+This makes it impossible to construct correct URLs on the server-side, as the protocol to use is not known.
+
+Even though not standard, the `X-Forwarded-*` family of HTTP headers can be used to preserve the details of the original connection and make them known to REANA's web server.
+To fix the particular issue presented before, it is important that proxies correctly set the `X-Forwarded-Proto` HTTP header.
+Note that the provided Traefik Ingress Controller acts as a proxy, but it correctly sets these headers by default.
+
+!!! warning
+
+    Using `X-Forwarded-*` headers can create security issues.
+    You must make sure that the `X-Forwarded-*` headers provided by the client are either discarded or overwritten by a proxy you trust.
+    See for example the [security concerns for X-Forwarded-For](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For#security_and_privacy_concerns).
+
+Let's consider another setup in which HAProxy sits in front of the provided Traefik instace, like this:
```

Comment:
Let us now consider another setup in which HAProxy sits in front of a separate Traefk instance:

(note "provided" vs "internal" Traefik, which could be again confusion-prone)

Reply:
In this context with "provided" I was referring to the "internal" Traefik that REANA deploys. The general idea also works for a "separate"/external instance of Traefik, but the configuration details might change a little bit depending on how that instance is deployed and configured. 

Should I perhaps remove "provided"?

_Let us now consider another setup in which HAProxy sits in front of Traefik: ..._

Reply:
If we use "provided", then we should probably list it under the "REANA" box, not as a separate box.

I see basically two solutions:

(1) We could change the use case to the one using "Manual ingress" and we say that REANA has internal ingress disabled and we put an extra Traefik instance before REANA; then the picture would be OK.

(2) We could keep the internal Traefik, but expose internal vs external Traefik situation clearer by painting a picture like:

```
           +--REANA---------------------+
           |            +----------+    |
Client --> | traefik -> | r-server |... |
           | enabled    +----------+    |
           +----------------------------+
```

This might perhaps overcomplicate things a bit, since we are starting to expose some of the internal components, but it would be closer to reality. And it would easily allow to explain also the other internal vs external traefik ingress use cases.

E.g. we could draw similar ASCII arts for all the sections above, showing this one for "Configuring default ingresses", whilst showing another one for "Setting up manual ingresses":

```
           +--External Ingress--+     +--REANA---------------------+
           |      traefik       |     |            +----------+    |
Client --> |       nginx        | --> | traefik -> | r-server |... |
           |        etc         |     | disabled   +----------+    |
           +--------------------+     +----------------------------+
```

Note also that people may be interested in how to use Nginx as ingress -- see https://github.com/reanahub/docs.reana.io/pull/148 -- which we may want to integrate into this page after merge. So it would be good to think about page structure accommodating nginx information in the near future.

OT1H, pictures of this kind would expose only some REANA internals such as r-ui or r-server, but that OTOH might be acceptable in a documentation page dedicated to ingresses. And we could expose r-w-controller, r-j-controller and friends in another page dedicated to the internal architecture.

If we go this way, then the ingress page ASCII art would be somewhat "deeper" than it currently is, but would still basically only "dip into" some of the REANA internals a bit, not bothering with anything which is not ingress related. (This would be exposed in another documentation page.) We could actually use dia/drawio/googleslides/etc to create these images instead of using ASCII art, so that they would have similar style in the "full picture" pages too.

(3) BTW, with the above in mind, the current nomenclature "Configuring default ingresses" vs "Setting up manual ingresses" might be perhaps better said "Configuring internal ingresses" vs "Configuring external ingresses"? Here I mean internal/external with respect to REANA, although people could think of "internal/external to cluster", which could be perhaps confusion-prone too.

Just some raw brain dump, we can chat IRL about pros/cons?

Reply:
I have pushed a new commit to track changes more easily, I will squash all of them when the PR is ready to be merged. I have also added one missing part for setting up manual ingresses.

Reply:
Good idea pushing a new commit and then squashing everything before merging :+1: The proxy text reads better now, I have left some cosmetic comments and mostly a section reordering comment.